builtin/logical/transit/path_derive_key.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package transit
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openbao/openbao/sdk/v2/framework"
+	"github.com/openbao/openbao/sdk/v2/helper/keysutil"
+	"github.com/openbao/openbao/sdk/v2/logical"
+)
+
+// The default key derivation algorithm is the cofactor Diffie-Hellman version of the elliptic curve key agreement scheme, as defined in ANSI X9.63
+const defaultKeyDerivationAlgorithm = "ecdh"
+
+func (b *backend) pathDeriveKey() *framework.Path {
+	return &framework.Path{
+		Pattern: "derive-key/" + framework.GenericNameRegex("name"),
+
+		DisplayAttrs: &framework.DisplayAttributes{
+			OperationPrefix: operationPrefixTransit,
+			OperationVerb:   "derive-key",
+		},
+
+		Fields: map[string]*framework.FieldSchema{
+
+			"key_derivation_algorithm": {
+				Type:    framework.TypeString,
+				Default: defaultKeyDerivationAlgorithm,
+				Description: `Key derivation algorithm to use. Valid values are:
+* ecdh
+Defaults to "ecdh".`,
+			},
+
+			"peer_public_key": {
+				Type:        framework.TypeString,
+				Description: "The pem-encoded other party's EC public key",
+			},
+
+			"base_key_name": {
+				Type:        framework.TypeString,
+				Description: "Name of the base key to use for derivation (own private key for ECDH)",
+			},
+
+			"base_key_version": {
+				Type: framework.TypeInt,
+				Description: `The version of the base key to use for derivation.
+Must be 0 (for latest) or a value greater than or equal
+to the min_derivation_version configured on the key.`,
+			},
+
+			"derived_key_name": {
+				Type:        framework.TypeString,
+				Description: "Name of the output derived key",
+			},
+
+			"derived_key_type": {
+				Type:    framework.TypeString,
+				Default: "aes256-gcm96",
+				Description: `
+The type of the output derived key. Currently, "aes128-gcm96" , "aes256-gcm96", "chacha20-poly1305", "xchacha20-poly1305" are supported.
+Defaults to "aes256-gcm96".
+`,
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
```

Comment:
Do you mind using `Operations:` ? `Callbacks:` is deprecated: https://github.com/openbao/openbao/blob/main/sdk/framework/path.go#L109

Reply:
This is addressed in #812 so there's no more examples of `Callbacks:`. :-)